Revisions to sns-asset-canister.md (#1631)
dfinity/portal: docs/developer-docs/integrations/sns/managing/sns-asset-canister.md

In the context of the SNS, a dapp's associated asset canister serves the frontend assets related to dapp and may also include a frontend to the SNS DAO, e.g., through which users can vote on governance proposals.  

The contents of the asset canister must be configured prior to the launch of the SNS, and any changes afterwards must be made by a principal with the `Prepare` permission. Principals with this `Prepare` permission can make a batch of changes to the asset canister and then 'lock' those changes. To have those changes applied, a proposal must be submitted. Anyone can submit the proposal for the 'locked' changes. Once changes are proposed, they can be voted on by the SNS DAO. If approved, the SNS governance canister is the only one that can commit the approved changes. This configuration assures that changes to the asset canister are only made by approved proposals. These changes are referred to as 'updates' to the asset canister in the remainder of this document. 

This section is relevant if your project contains an asset canister and describes how you can test handing over control of an asset canister to an SNS.

:::info 

> In this document, the term **upgrade** refers to deploying an upgraded version of a canister's Wasm file.

Reviewer: ```suggestion
In this document, the term **upgrade** refers to deploying an upgraded version of a canister's WebAssembly module.
```

> The term **update** refers to changing or updating the assets stored within an SNS asset canister.

Reviewer: ```suggestion
The term **update** refers to changing or updating the assets stored within an asset canister.
```

The asset canister is not an SNS canister (such as governance, root etc.). It can be _controlled_ by SNS but the notion of _update_ applies also to asset canisters not controlled by an SNS.

    - The user or developer creating the SNS should remove their own personal permissions. 
- Lastly, the SNS's function should be registered to commit the configuration.

To deploy an asset canister, any canister in the `dfx.json` file can be set as `"type": "assets"`, and dfx will automatically generate the required files, such as an assets.wasm.gz file and a candid/assets.did file once the canister has been built. 

An example of configuring an asset canister within the `dfx.json` file can be found below:

```
{
  "canisters": {

>     "assets": {

Reviewer: In addition to setting the "type" field, I'd also drop the remaining fields and use something like this:
```
    "assets": {
      "source": [],
      "type": "assets"
    }
```

Once the asset canister has been handed over to the SNS, only the governance canister should have `Commit` rights, and principals in the whitelist should have `Prepare` rights. The developer who configured and deployed the SNS should have their permissions removed during the SNS launch. 

> If the whitelist needs to be adapted or changed, it is recommended that the SNS calls `take_ownership` via a proposal, which must be added as a custom proposal type. This will clear all permissions, and give only the SNS governance canister `Commit` permissions. If the SNS does not call `take_ownership`, a user cannot verify that all changes to the assets have been approved by an SNS proposal. 

Reviewer: ```suggestion
It is recommended that the SNS calls `take_ownership` via a proposal, which must be added as a custom proposal type. This will clear all permissions, and give only the SNS governance canister `Commit` permissions. If the SNS does not call `take_ownership`, a user cannot verify that all changes to the assets have been approved by an SNS proposal. 
```
